packages/rpc-methods/src/restricted/manageState.test.ts

```diff
@@ -85,19 +183,66 @@ describe('snap_manageState', () => {
         clearSnapState,
         getSnapState,
         updateSnapState,
+        getMnemonic: jest.fn().mockResolvedValue(TEST_SECRET_RECOVERY_PHRASE),
+        getUnlockPromise: jest.fn(),
+      });
+
+      await manageStateImplementation({
+        context: { origin: MOCK_SNAP_ID },
+        method: 'snap_manageState',
+        params: {
+          operation: ManageStateOperation.UpdateState,
+          newState: mockSnapState,
+        },
       });
-      const newState = { data: 'updated data' };
 
       await manageStateImplementation({
-        context: { origin: 'snap-origin' },
+        context: { origin: MOCK_LOCAL_SNAP_ID },
         method: 'snap_manageState',
         params: {
           operation: ManageStateOperation.UpdateState,
-          newState,
+          newState: mockSnapState,
         },
       });
 
-      expect(updateSnapState).toHaveBeenCalledWith('snap-origin', newState);
+      expect(updateSnapState).toHaveBeenCalledTimes(2);
+      expect(updateSnapState).toHaveBeenNthCalledWith(
+        1,
+        MOCK_SNAP_ID,
+        mockEncryptedState,
+      );
+
+      expect(updateSnapState).not.toHaveBeenNthCalledWith(
```

Comment:
Could we verify that it has been called with the expected encrypted state and that it is different? This assertion also passes if it isn't called at all

Reply:
It's asserted that the function is called exactly twice.

Reply:
Ah that is true, but I would still be more comfortable with us asserting that the parameters match what we expect